src/dds/rs-dds-sensor-proxy.cpp

```diff
@@ -286,15 +288,21 @@ void dds_sensor_proxy::open( const stream_profiles & profiles )
 
 
 void dds_sensor_proxy::handle_video_data( realdds::topics::image_msg && dds_frame,
+                                          realdds::dds_sample && dds_sample,
                                           const std::shared_ptr< stream_profile_interface > & profile,
                                           streaming_impl & streaming )
 {
     frame_additional_data data;  // with NO metadata by default!
+    data.system_time = time_service::get_time();  // time of arrival in system clock
+    data.backend_timestamp                        // time when the underlying backend (DDS) received it
+        = static_cast<rs2_time_t>(realdds::time_to_double( dds_sample.reception_timestamp ) * 1e3);
     data.timestamp               // in ms
         = static_cast< rs2_time_t >( realdds::time_to_double( dds_frame.timestamp ) * 1e3 );
```

Comment:
What is `dds_frame.timestamp` source, the time when it was sent from the camera? Shouldn't we adjust it to our clock?

Reply:
`dds_frame.timestamp` is the timestamp in the `sensor_msgs::msg::Image` header. The camera decides on the frame timestamp and this is also what's used to synchronize with the metadata.